core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -161,6 +165,82 @@ static ClientRequestContextBuilder builder(RpcRequest request, URI uri) {
         return new ClientRequestContextBuilder(request, uri);
     }
 
+    /**
+     * Returns the {@link ServiceRequestContext} whose {@link Service} invokes the {@link Client}
+     * {@link Request} which created this {@link ClientRequestContext}, or {@code null} if this client request
+     * was not made in the context of a server request.
+     */
+    @Nullable
+    ServiceRequestContext rootContext();
```

Comment:
I guess the main reason for exposing `ServiceRequestContext` is to get attributes from the `rootContext`. Then don't we change `ServiceRequestContext` to `RequestContext`?
If I misunderstood, feel free to ignore this review. 😀

Reply:
It's for context aware and for https://github.com/line/armeria/issues/2173#issuecomment-559438479 🙏

Reply:
Thanks for the link.